src/pages/TermsOfService.js

```diff
@@ -1,281 +1,341 @@
+Terra is not a Covered Entity as that term is defined in the Health
+Insurance Portability and Accountability Act of 1996, as amended, and
+its related regulations (collectively, "HIPAA"). On occasion, Terra
+may agree in writing with a user to perform services for the user in the
+storing PHI. We recommend that such users enter into a formal agreement
+with Terra/Firecloud.
+
+Terra does offer clinical compliance features as part of its service
+("Compliance Features") for users who wish to upload, store, or
+otherwise transfer PHI, as well as users who are using the Site in
+connection with their clinical operations. Users who desire to upload,
+store, or otherwise transfer PHI using the Site must implement all of
+the required Clinical Features and must enter into a formal agreement
+with Terra/Firecloud stating that. The uploading, storing, or
+transferring of PHI using the Site by users that have not implemented
+the Clinical Features is strictly prohibited. You agree that, unless you
+have implemented the Clinical Features, you will not upload, store, or
+otherwise transfer PHI using the Site.
+
+You acknowledge that this may require you, in some instances, to
+anonymize sequence data prior to uploading it to the Site. You further
+agree to indemnify and hold harmless Terra of and from any and all
+claims, demands, losses, causes of action, damage, lawsuits, judgments,
+including attorneys' fees and costs, arising out of or relating to your
+uploading, storing, or transferring of PHI without having fully
+implemented the Clinical Features.
+
+Terms as of February 10, 2020.
 `
 
-const styles = {
-  page: {
-    padding: '1rem', minHeight: '100%',
-    display: 'flex', justifyContent: 'center', alignItems: 'center'
-  },
-  box: {
-    backgroundColor: 'rgba(255,255,255,0.85)',
-    borderRadius: 5, width: 800, padding: '2rem', boxShadow: Style.standardShadow
-  }
-}
-
-const termsTitle = div({ style: { color: colors.dark(), fontWeight: 600 } }, [
-  span({ style: { fontSize: 36 } }, ['TERRA ']), span({ style: { fontSize: 24 } }, ['Terms of Service'])
-])
-
-const TOSMarkdown = div({
-  style: { maxHeight: 400, overflowY: 'auto', lineHeight: 1.5, marginTop: '1rem', paddingRight: '1rem' }
-}, [
-  h(MarkdownViewer, {
-    renderers: {
-      heading: (text, level) => {
-        return `<h${level} style="color: ${colors.dark()}; margin-bottom: 0">${text}</h${level}>`
-      },
-      paragraph: text => {
-        return `<p style="margin-top: 0">${text}</p>`
-      }
-    }
-  }, [termsOfService])
-])
-
-export default class TermsOfService extends Component {
-  constructor(props) {
-    super(props)
-    this.state = { busy: false }
-  }
+const TermsOfServicePage = () => {
+  const [busy, setBusy] = useState()
+  const { isSignedIn, acceptedTos } = authStore.get() // can't change while viewing this without causing it to unmount, so doesn't need to subscribe
 
-  async accept() {
+  const accept = async () => {
```

Comment:
Can/should this use `Utils.withBusyState`?

Reply:
That would cause a state change after unmount in the success case.

Reply:
Cool. At some point, could you teach me more about this? It's not quite clicking for me.